smac/optimizer/smbo.py

```diff
@@ -198,14 +198,19 @@ def run(self) -> Configuration:
 
             # sample next configuration for intensification
             # Initial design runs are also included in the BO loop now.
-            run_info = self.intensifier.get_next_run(
+            behest, run_info = self.intensifier.get_next_run(
                 challengers=self.initial_design_configs,
                 incumbent=self.incumbent,
                 chooser=self.epm_chooser,
                 run_history=self.runhistory,
                 repeat_configs=self.intensifier.repeat_configs,
             )
 
+            # Check if no more configs,
+            if behest == IntensifierBehest.TERMINATE:
+                self.logger.warn("No more valid configurations to try during intensification")
+                break
```

Comment:
It should be a break, because there are no more configurations to run. 

That is, in the unlikely event (but possible in unit testing due to the limited configs to try) that there is nothing more to try, this intent is triggered

Reply:
This still puzzles me. Having a break here completely ends the optimization and does not move to a new iteration. However, we did not have such a feature before. Maybe we need to discuss this offline.